### PR TITLE
Толстиков Максим - Лабораторная работа #2

### PR DESCRIPTION
**Pull request by Tolstikov98**

### CODE REVIEW

**Comment by Tolstikov98** [comment]
@alexsoll 
@VadimKutovoi

File: modules/vector-3d/test/test_vector_3d.cpp [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+TEST_F(Vector3dTest, Number_Is_Equal_To_Itself) {
+    // Arrange
+    Vector3d v(26.0, 14.0, 5.0);
+
+    // Act & Assert
+    EXPECT_TRUE(v == v);
+}
+
+TEST_F(Vector3dTest, Equal_Numbers_Are_Equal) {
```
Comment (VadimKutovoi) [inline modules/vector-3d/test/test_vector_3d.cpp]: I think you mean Equal_Vectors_Are_Equal

File: modules/vector-3d/test/test_vector_3d.cpp [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+    // Arrange
+    Vector3d v1(26.0, 14.0, 5.0);
+    Vector3d v2(26.0, 14.0, 5.0);
+
+    // Act & Assert
+    EXPECT_EQ(v1, v2);
+}
+
+TEST_F(Vector3dTest, Different_Numbers_Not_Equal) {
```
Comment (VadimKutovoi) [inline modules/vector-3d/test/test_vector_3d.cpp]: Same

File: modules/vector-3d/include/vector_3d.h [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    double Norma() const;
+    Vector3d Normalize() const;
+
+    bool operator == (const Vector3d& v) const;
+    bool operator != (const Vector3d& v) const;
+
+ private:
+    double x_;
+    double y_;
+    double z_;
```
Comment (VadimKutovoi) [inline modules/vector-3d/include/vector_3d.h]: Is it necessary to use underscore after variable name?

File: modules/vector-3d/test/test_vector_3d.cpp [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+    // Act
+    Vector3d v = v2 - v1;
+
+    // Assert
+    Vector3d expected_v(-25.0, -13.0, -4.0);
+    EXPECT_EQ(expected_v, v);
+}
+
+TEST_F(Vector3dTest, Can_Multiplicate_Scalar) {
```
Comment (VadimKutovoi) [inline modules/vector-3d/test/test_vector_3d.cpp]: Maybe correct it to Can_Multiplicate_By_Scalar like test name below?

File: modules/vector-3d/include/vector_3d.h [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+#ifndef MODULES_VECTOR_3D_INCLUDE_VECTOR_3D_H_
+#define MODULES_VECTOR_3D_INCLUDE_VECTOR_3D_H_
+
+class Vector3d {
+ public:
+    Vector3d();
+    Vector3d(const double x, const double y, const double z);
+    Vector3d(const Vector3d& v);
+
+    Vector3d& operator=(const Vector3d& v);
```
Comment (VadimKutovoi) [inline modules/vector-3d/include/vector_3d.h]: In my opinion it's better to move all operator's overloading to one "section" and add whitespaces on the left and on the right of operator or remove them for all operators. Also please correct it to functions definitions of operator function.

File: modules/vector-3d/include/vector_3d.h [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    Vector3d operator + (const Vector3d& v) const;
+    Vector3d operator - (const Vector3d& v) const;
+    Vector3d operator * (const double& d) const;
+    Vector3d operator / (const double& d) const;
+    double operator * (const Vector3d& v) const;
+    Vector3d Cross(const Vector3d& v) const;
+    double Norma() const;
+    Vector3d Normalize() const;
+
+    bool operator == (const Vector3d& v) const;
```
Comment (VadimKutovoi) [inline modules/vector-3d/include/vector_3d.h]: Same, move to one section.

File: modules/vector-3d/src/vector_3d.cpp [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+void Vector3d::setY(const double y) {
+    y_ = y;
+}
+
+void Vector3d::setZ(const double z) {
+    z_ = z;
+}
+
+Vector3d Vector3d::operator+(const Vector3d& v) const {
```
Comment (VadimKutovoi) [inline modules/vector-3d/src/vector_3d.cpp]: Same thing as in my first comment. Correct it to all definitions, please.

File: modules/vector-3d/test/test_vector_3d.cpp (hunk 1 above)
Comment (Tolstikov98) [reply modules/vector-3d/test/test_vector_3d.cpp]: Yes sir!

File: modules/vector-3d/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,13 @@
+# Declare variables for binaries' names
+get_filename_component(DIR_NAME ${CMAKE_CURRENT_LIST_DIR} NAME)
+set(MODULE      "${DIR_NAME}")
+set(LIBRARY     "lib_${MODULE}")
+set(TESTS       "test_${MODULE}")
+set(APPLICATION "app_${MODULE}")
+
+# Include directory with public headers
+include_directories("./")
```
Comment (alexbatashev) [inline modules/vector-3d/CMakeLists.txt]: CMAKE_CURRENT_SOURCE_DIR

File: modules/vector-3d/test/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,27 @@
+set(target ${TESTS})
+
+file(GLOB srcs "*.cpp")
+set_source_files_properties(${srcs} PROPERTIES
+    LABELS "${MODULE};Test")
+
+# VS2012 doesn't support correctly the tuples yet,
+# see http://code.google.com/p/googletest/issues/detail?id=412
+if(MSVC)
```
Comment (alexbatashev) [inline modules/vector-3d/test/CMakeLists.txt]: Do you really need this?

File: modules/vector-3d/test/test_vector_3d.cpp [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+
+    // Act
+    Vector3d v = v2 - v1;
+
+    // Assert
+    Vector3d expected_v(-25.0, -13.0, -4.0);
+    EXPECT_EQ(expected_v, v);
+}
+
+TEST_F(Vector3dTest, Can_Multiplicate_By_Scalar) {
```
Comment (alexbatashev) [inline modules/vector-3d/test/test_vector_3d.cpp]:
```suggestion
TEST_F(Vector3dTest, Can_Multipliy_By_Scalar) {
```

File: modules/vector-3d/test/test_vector_3d.cpp [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+
+    // Act
+    double d = v1 * v2;
+
+    // Assert
+    double expected_d = 32.0;
+    EXPECT_EQ(expected_d, d);
+}
+
+TEST_F(Vector3dTest, Can_Multiplicate_Vector) {
```
Comment (alexbatashev) [inline modules/vector-3d/test/test_vector_3d.cpp]:
```suggestion
TEST_F(Vector3dTest, Can_Multipliy_Vector) {
```

File: modules/vector-3d/test/test_vector_3d.cpp [hunk 12]
```diff
@@ -0,0 +1,230 @@
...
+
+    // Act
+    Vector3d v = v1.Cross(v2);
+
+    // Assert
+    Vector3d expected_v(2.0, -13.0, 8.0);
+    EXPECT_EQ(expected_v, v);
+}
+
+TEST_F(Vector3dTest, Can_Find_Norma) {
```
Comment (alexbatashev) [inline modules/vector-3d/test/test_vector_3d.cpp]: Did you mean Norm?

File: modules/vector-3d/test/test_vector_3d.cpp (hunk 10 above)
Comment (Tolstikov98) [reply modules/vector-3d/test/test_vector_3d.cpp]: Perhaps better to write Can_Multiply_By_Scalar.

**Comment by Tolstikov98** [comment]
@alexbatashev

**Pull request merged**